src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/InMemoryPartitionedAppTokenCacheAccessor.cs

```diff
@@ -216,13 +234,25 @@ public void SetiOSKeychainSecurityGroup(string keychainSecurityGroup)
         public virtual void Clear()
         {
             AccessTokenCacheDictionary.Clear();
-            _logger.Always("[Clear] Clearing access token cache data.");
+            Interlocked.Exchange(ref _cacheSize, 0);
+            _logger.Always("[AppCache] Cleared access token cache data.");
             // app metadata isn't removable
         }
 
         public virtual bool HasAccessOrRefreshTokens()
         {
             return AccessTokenCacheDictionary.Any(partition => partition.Value.Any(token => !token.Value.IsExpiredWithBuffer()));
         }
+
+        private bool IsOverCapacity()
+        {
+            return _tokenCacheAccessorOptions.SizeLimit.HasValue && (Interlocked.Read(ref _cacheSize) + ATSizeInBytes) > _tokenCacheAccessorOptions.SizeLimit;
+        }
+
+        private void Compact()
```

Comment:
Do we need Compact method? It just calls clear. (logging can be done by the caller.)

Reply:
Called it `Compact` as a form of abstraction, clearing is just an implementation.

Reply:
I agree with Peter using Compact() instead of Clear() directly.